pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyClientCnx.java

```diff
@@ -54,10 +67,33 @@ protected ByteBuf newConnectCommand() throws Exception {
 
         authenticationDataProvider = authentication.getAuthData(remoteHostName);
         AuthData authData = authenticationDataProvider.authenticate(AuthData.INIT_AUTH_DATA);
-        return Commands.newConnect(authentication.getAuthMethodName(), authData, this.protocolVersion,
-            PulsarVersion.getVersion(), proxyToTargetBrokerAddress, clientAuthRole, clientAuthData,
-            clientAuthMethod);
+        return Commands.newConnect(authentication.getAuthMethodName(), authData, protocolVersion,
+                PulsarVersion.getVersion(), proxyToTargetBrokerAddress, clientAuthRole, clientAuthData,
+                clientAuthMethod);
     }
 
-    private static final Logger log = LoggerFactory.getLogger(ProxyClientCnx.class);
+    @Override
+    protected void handleAuthChallenge(CommandAuthChallenge authChallenge) {
+        checkArgument(authChallenge.hasChallenge());
+        checkArgument(authChallenge.getChallenge().hasAuthData());
+
+        boolean isRefresh = Arrays.equals(AuthData.REFRESH_AUTH_DATA_BYTES, authChallenge.getChallenge().getAuthData());
+        if (!forwardClientAuthData || !isRefresh || refreshClientAuthDataNotifier == null) {
+            super.handleAuthChallenge(authChallenge);
+            return;
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Request to refresh the original client authentication data");
+        }
+        try {
+            refreshClientAuthDataNotifier.get();
+            if (state == State.SentConnectFrame) {
+                state = State.Connecting;
```

Comment:
Does this state conflict with `handleAuthResponse`? Because I see the logic check 
```
  // First connection
if (this.connectionPool == null || state == State.Connecting) {
     // authentication has completed, will send newConnected command.
      completeConnect(clientData);
} 
```

Reply:
No conflicts, one is from `ProxyConnection`, and one is from `ProxyClientCnx`.